fieldpath/element.go

```diff
@@ -24,6 +24,58 @@ import (
 	"sigs.k8s.io/structured-merge-diff/value"
 )
 
+type KeyValue struct {
```

Comment:
Is this the same as value.Field? https://github.com/kubernetes-sigs/structured-merge-diff/blob/master/value/value.go#L127 Can we reuse that?